engine/buff/buff.cpp

```diff
@@ -1516,6 +1588,10 @@ buff_t* buff_t::apply_affecting_effect( const spelleffect_data_t& effect )
         apply_percent_modifier( effect );
         break;
 
+      case A_MOD_TIME_RATE:
+        set_time_modifier( effect.base_value() );
+        break;
+
```

Comment:
This is not the right place to automatically parse this aura type. A_MOD_TIME_RATE is global while `spell_data_t::affected_by( const spelleffect_data_t* effect )` is for checking if the spell's class flags match the ones in the effect.

If supporting this aura type is needed, it would make more sense to add a virtual function like `player_t::composite_time_rate_multiplier` and something that can be called to adjust buffs when changing it similar to how `action_t::recharge_rate_multiplier` is used with `cooldown_t::adjust_recharge_multiplier` to adjust cooldowns.

For now, it looks like the Rogue changes only need A_MOD_TIME_RATE_BY_SPELL_LABEL (and it only needs to apply to the buff duration), so it probably makes more sense to leave this out for now until it is needed since fully supporting the aura type would also require adjusting many other parts of the actor (attack speed, movement speed, cooldown recharge rate, etc.) and Blizzard has recently avoided using A_MOD_TIME_RATE in relevant content outside of occasionally pausing an encounter briefly with a -99% aura.